### DISCUSSION TITLE
ZAP vignette: probability of observing

### DISCUSSION THREAD

**Discussion by dmi3kno** [inlabru-org/inlabru#216] [Q&A] [answered]
I was reflecting on the new ZAP vignette and how the model could be validated on the new data (holdout set).

In lines 328-329 we have:

https://github.com/inlabru-org/inlabru/blob/77c82470d13930d7d542b28c2f3607143db33a07/vignettes/articles/zip_zap_models.Rmd#L328-L330

Here `count` comes from the data, not from predictions. In the holdout set, we would not have that value. Instead, we would have to rely on `lambda*area`. I think. Perhaps, it would be safer to just use the predicted count instead of the (known) actual count?

Perhaps it would be a good idea to set aside a portion of data as a holdout? The area of the holdout set would have to be covered by the model mesh (otherwise spatial interpolation would not work), but we can not include the datapoints in the training set for the holdout location, correct?

**Comment by finnlindgren** [comment, marked as answer]
The quoted code computes the probability of a specific `count` value, whether it's in the training data or a holdout data set. The difference is whether that count was also used in the estimation of `lambda` or not, i.e. in the training set we're computing "in-sample" predictions and for the hold-out set it's "out-of-sample" predictions.

Setting aside holdout data to assess prediction more fairly than in-sample assessment is sensible, but in the spatial setting it's not as simple as in the "independent samples" setting, so various strategies have been and are still being developed. For example, group-wise cross validation approaches.
